### PR TITLE
Correct `Port.info/1,2` functions

### PR DESCRIPTION
**Pull request by lexmag**
Hey, I would appreciate any feedback on this.

### CODE REVIEW

File: lib/elixir/lib/port.ex [hunk 1]
```diff
@@ -58,21 +92,33 @@ defmodule Port do
...
-  See http://www.erlang.org/doc/man/erlang.html#port_info-1.
+  Returns information about the `port`
+  or `nil` if the port is closed.
 
-  Inlined by the compiler.
+  See http://www.erlang.org/doc/man/erlang.html#port_info-1.
   """
   def info(port) do
-    :erlang.port_info(port)
+    nillify :erlang.port_info(port)
```
Comment (lexmag) [inline lib/elixir/lib/port.ex]: @josevalim please note that this is a breaking change (same for `info/2`).
Would you like me to extract it to the separate commit?

Comment (josevalim) [inline lib/elixir/lib/port.ex]: It is ok to be here. It is a bug fix if it is inconsistent with Process.

**Comment by lexmag** [comment]
Updated. I left `Process.link/1` mentioning `port` since it is still a valid way to call it. :bowtie:

**Comment by lexmag** [comment]
@josevalim this is ready for merge. Could you take a look on it?

File: lib/elixir/lib/port.ex [hunk 2]
```diff
@@ -22,6 +22,40 @@ defmodule Port do
...
   @doc """
+  Associates the name with a port identifier. `name`, which must
+  be an atom, can be used instead of the port identifier with the
+  `Kernel.send/2` function.
+
+  `Port.register/2` will fail with `ArgumentError` if the port supplied
+  is closed, or if the name is already registered (check with `whereis/1`).
+  """
+  @spec register(port, atom) :: true
+  def register(port, name) when not name in [nil, false, true] do
```
Comment (josevalim) [inline lib/elixir/lib/port.ex]: Should we check if the first argument is actually a port? This patch has me a bit torn because the stuff in Process will mostly work with ports too.

File: lib/elixir/lib/process.ex [hunk 3]
```diff
@@ -121,7 +121,7 @@ defmodule Process do
 
   """
   @spec send(dest, msg, [option]) ::  result when
-        dest: pid | port | atom | {atom, node},
+        dest: pid | atom | {atom, node},
```
Comment (josevalim) [inline lib/elixir/lib/process.ex]: We should keep port here because we can send messages to port.

**Comment by lexmag** [comment]
Updated.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
